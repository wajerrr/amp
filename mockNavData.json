{
  "navigation": {
    "headline": "test-menu-amp",
    "hasPart": {
      "parts": [
        {
          "headline": "Topics",
          "id": "21741809",
          "hasPart": {
            "parts": [
              {
                "headline": "Latest updates",
                "id": "storyCollection_21741809_0",
                "url": {
                  "canonical": "https://www.economist.com/latest-updates"
                }
              },
              {
                "headline": "Leaders",
                "id": "storyCollection_21741809_1",
                "url": {
                  "canonical": "https://www.economist.com/sections/leaders"
                }
              },
              {
                "headline": "Briefing",
                "id": "storyCollection_21741809_2",
                "url": {
                  "canonical": "https://www.economist.com/sections/briefings"
                }
              },
              {
                "headline": "United States",
                "id": "storyCollection_21741809_3",
                "url": {
                  "canonical": "https://www.economist.com/sections/united-states"
                }
              },
              {
                "headline": "The Americas",
                "id": "storyCollection_21741809_4",
                "url": {
                  "canonical": "https://www.economist.com/sections/americas"
                }
              },
              {
                "headline": "Asia",
                "id": "storyCollection_21741809_5",
                "url": {
                  "canonical": "https://www.economist.com/sections/asia"
                }
              },
              {
                "headline": "China",
                "id": "storyCollection_21741809_6",
                "url": {
                  "canonical": "https://www.economist.com/sections/china"
                }
              },
              {
                "headline": "Middle East and Africa",
                "id": "storyCollection_21741809_7",
                "url": {
                  "canonical": "https://www.economist.com/sections/middle-east-africa"
                }
              },
              {
                "headline": "Europe",
                "id": "storyCollection_21741809_8",
                "url": {
                  "canonical": "https://www.economist.com/sections/europe"
                }
              },
              {
                "headline": "Britain",
                "id": "storyCollection_21741809_9",
                "url": {
                  "canonical": "https://www.economist.com/sections/britain"
                }
              },
              {
                "headline": "International",
                "id": "storyCollection_21741809_10",
                "url": {
                  "canonical": "https://www.economist.com/sections/international"
                }
              },
              {
                "headline": "Business",
                "id": "storyCollection_21741809_11",
                "url": {
                  "canonical": "https://www.economist.com/sections/business-finance"
                }
              },
              {
                "headline": "Finance and economics",
                "id": "storyCollection_21741809_12",
                "url": {
                  "canonical": "https://www.economist.com/sections/economics"
                }
              },
              {
                "headline": "Science and technology",
                "id": "storyCollection_21741809_13",
                "url": {
                  "canonical": "https://www.economist.com/sections/science-technology"
                }
              },
              {
                "headline": "Books and arts",
                "id": "storyCollection_21741809_14",
                "url": {
                  "canonical": "https://www.economist.com/sections/culture"
                }
              },
              {
                "headline": "Obituary",
                "id": "storyCollection_21741809_15",
                "url": {
                  "canonical": "https://www.economist.com/sections/obituary"
                }
              },
              {
                "headline": "Special reports",
                "id": "storyCollection_21741809_16",
                "url": {
                  "canonical": "https://www.economist.com/sections/special-reports"
                }
              },
              {
                "headline": "Technology Quarterly",
                "id": "storyCollection_21741809_17",
                "url": {
                  "canonical": "https://www.economist.com/sections/technology-quarterly"
                }
              },
              {
                "headline": "Debates",
                "id": "storyCollection_21741809_18",
                "url": {
                  "canonical": "http://debates.economist.com/"
                }
              }
            ]
          }
        },
        {
          "headline": "Blogs",
          "id": "storyCollection_21741814_21741810",
          "canonical": "/node/21741810",
          "originalID": "21741810",
          "hasPart": {
            "parts": [
              {
                "headline": "Bagehot’s notebook",
                "id": "storyCollection_21741810_0",
                "url": {
                  "canonical": "https://www.economist.com/blogs/bagehot"
                }
              },
              {
                "headline": "Buttonwood’s notebook",
                "id": "storyCollection_21741810_1",
                "url": {
                  "canonical": "https://www.economist.com/blogs/buttonwood"
                }
              },
              {
                "headline": "Democracy in America",
                "id": "storyCollection_21741810_2",
                "url": {
                  "canonical": "https://www.economist.com/blogs/democracyinamerica"
                }
              },
              {
                "headline": "Erasmus",
                "id": "storyCollection_21741810_3",
                "url": {
                  "canonical": "https://www.economist.com/blogs/erasmus"
                }
              },
              {
                "headline": "Free exchange",
                "id": "storyCollection_21741810_4",
                "url": {
                  "canonical": "https://www.economist.com/blogs/freeexchange"
                }
              },
              {
                "headline": "Game theory",
                "id": "storyCollection_21741810_5",
                "url": {
                  "canonical": "https://www.economist.com/blogs/gametheory"
                }
              },
              {
                "headline": "Graphic detail",
                "id": "storyCollection_21741810_6",
                "url": {
                  "canonical": "https://www.economist.com/blogs/graphicdetail"
                }
              },
              {
                "headline": "Gulliver",
                "id": "storyCollection_21741810_7",
                "url": {
                  "canonical": "https://www.economist.com/blogs/gulliver"
                }
              },
              {
                "headline": "Kaffeeklatsch",
                "id": "storyCollection_21741810_8",
                "url": {
                  "canonical": "https://www.economist.com/blogs/kaffeeklatsch"
                }
              },
              {
                "headline": "Prospero",
                "id": "storyCollection_21741810_9",
                "url": {
                  "canonical": "https://www.economist.com/blogs/prospero"
                }
              },
              {
                "headline": "The Economist explains",
                "id": "storyCollection_21741810_10",
                "url": {
                  "canonical": "https://www.economist.com/blogs/economist-explains"
                }
              }
            ]
          }
        },
        {
          "headline": "Current edition",
          "id": "storyCollection_21741814_2",
          "url": {
            "canonical": "https://www.economist.com/printedition"
          }
        },
        {
          "headline": "Audio edition",
          "id": "storyCollection_21741814_3",
          "url": {
            "canonical": "https://www.economist.com/audio-edition"
          }
        },
        {
          "headline": "Economist Films",
          "id": "storyCollection_21741814_4",
          "url": {
            "canonical": "http://films.economist.com/"
          }
        },
        {
          "headline": "Economist Radio",
          "id": "storyCollection_21741814_5",
          "url": {
            "canonical": "http://radio.economist.com/"
          }
        },
        {
          "headline": "The Economist apps",
          "id": "storyCollection_21741814_6",
          "url": {
            "canonical": "https://www.economist.com/digital"
          }
        },
        {
          "headline": "More",
          "id": "storyCollection_21741814_21741813",
          "hasPart": {
            "parts": [
              {
                "headline": "Apps & Digital Editions",
                "id": "storyCollection_21741813_21741811",
                "canonical": "/node/21741811",
                "hasPart": {
                  "parts": [
                    {
                      "headline": "The Economist apps",
                      "id": "storyCollection_21741811_0",
                      "url": {
                        "canonical": "https://www.economist.com/digital"
                      }
                    },
                    {
                      "headline": "Espresso",
                      "id": "storyCollection_21741811_1",
                      "url": {
                        "canonical": "https://www.economist.com/digital"
                      }
                    },
                    {
                      "headline": "Global Business Review",
                      "id": "storyCollection_21741811_2",
                      "url": {
                        "canonical": "https://www.economist.com/digital"
                      }
                    },
                    {
                      "headline": "World in Figures",
                      "id": "storyCollection_21741811_3",
                      "url": {
                        "canonical": "https://worldinfigures.com/"
                      }
                    }
                  ]
                }
              },
              {
                "headline": "From The Economist Group",
                "id": "storyCollection_21741813_21741812",
                "hasPart": {
                  "parts": [
                    {
                      "headline": "Events",
                      "id": "storyCollection_21741812_0",
                      "url": {
                        "canonical": "https://www.economist.com/events-conferences"
                      }
                    },
                    {
                      "headline": "Online GMAT prep",
                      "id": "storyCollection_21741812_1",
                      "url": {
                        "canonical": "https://gmat.economist.com/?gsrc=economist_nav"
                      }
                    },
                    {
                      "headline": "Online GRE prep",
                      "id": "storyCollection_21741812_2",
                      "url": {
                        "canonical": "https://gre.economist.com/?gsrc=economist_nav"
                      }
                    },
                    {
                      "headline": "Executive Education Navigator",
                      "id": "storyCollection_21741812_3",
                      "url": {
                        "canonical": "https://execed.economist.com/"
                      }
                    },
                    {
                      "headline": "Which MBA",
                      "id": "storyCollection_21741812_4",
                      "url": {
                        "canonical": "https://www.economist.com/whichmba"
                      }
                    },
                    {
                      "headline": "Jobs Board",
                      "id": "storyCollection_21741812_5",
                      "url": {
                        "canonical": "http://jobs.economist.com/"
                      }
                    },
                    {
                      "headline": "Learning.ly",
                      "id": "storyCollection_21741812_6",
                      "url": {
                        "canonical": "https://learning.ly/"
                      }
                    },
                    {
                      "headline": "The Economist Store",
                      "id": "storyCollection_21741812_7",
                      "url": {
                        "canonical": "https://shop.economist.com/"
                      }
                    },
                    {
                      "headline": "The Economist Intelligence Unit",
                      "id": "storyCollection_21741812_8",
                      "url": {
                        "canonical": "https://www.eiu.com/"
                      }
                    },
                    {
                      "headline": "The Economist Corporate Network",
                      "id": "storyCollection_21741812_9",
                      "url": {
                        "canonical": "https://www.corporatenetwork.com/"
                      }
                    }
                  ]
                }
              },
              {
                "headline": "About The Economist",
                "id": "storyCollection_21741813_2",
                "url": {
                  "canonical": "https://www.economist.com/about-the-economist"
                }
              }
            ]
          }
        },
        {
          "headline": "Subscribe",
          "id": "storyCollection_21741814_8",
          "url": {
            "canonical": "https://econ.st/2bs1gXz"
          }
        }
      ]
    }
  }
}
