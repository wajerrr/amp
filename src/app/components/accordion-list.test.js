import React from 'react';
import renderer from 'react-test-renderer';
import AccordionListItem from './accordion-list';
import * as navigationData from '../../../mockNavData.json';

describe('AccordionListItem', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(<AccordionListItem item={navigationData.navigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot with subscribeButton', () => {
    const tree = renderer
      .create(<AccordionListItem item={{ headline: 'Subscribe' }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot with normal link', () => {
    const tree = renderer
      .create(<AccordionListItem item={{ headline: 'Link' }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
