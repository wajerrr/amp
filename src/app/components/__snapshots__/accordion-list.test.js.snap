// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionListItem should match snapshot 1`] = `
<amp-accordion
  id="accordion"
>
  <section
    className="css-w82pwy"
  >
    <h4
      className="css-19ey912"
    >
      test-menu-amp
    </h4>
    <div>
      <amp-accordion
        id="accordion"
      >
        <section
          className="css-w82pwy"
        >
          <h4
            className="css-19ey912"
          >
            Topics
          </h4>
          <div>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Latest updates
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Leaders
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Briefing
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              United States
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              The Americas
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Asia
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              China
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Middle East and Africa
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Europe
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Britain
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              International
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Business
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Finance and economics
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Science and technology
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Books and arts
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Obituary
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Special reports
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Technology Quarterly
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Debates
            </a>
          </div>
        </section>
      </amp-accordion>
      <amp-accordion
        id="accordion"
      >
        <section
          className="css-w82pwy"
        >
          <h4
            className="css-19ey912"
          >
            Blogs
          </h4>
          <div>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Bagehot’s notebook
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Buttonwood’s notebook
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Democracy in America
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Erasmus
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Free exchange
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Game theory
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Graphic detail
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Gulliver
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Kaffeeklatsch
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              Prospero
            </a>
            <a
              className="css-1xnkqks"
              href="/"
            >
              The Economist explains
            </a>
          </div>
        </section>
      </amp-accordion>
      <a
        className="css-1xnkqks"
        href="/"
      >
        Current edition
      </a>
      <a
        className="css-1xnkqks"
        href="/"
      >
        Audio edition
      </a>
      <a
        className="css-1xnkqks"
        href="/"
      >
        Economist Films
      </a>
      <a
        className="css-1xnkqks"
        href="/"
      >
        Economist Radio
      </a>
      <a
        className="css-1xnkqks"
        href="/"
      >
        The Economist apps
      </a>
      <amp-accordion
        id="accordion"
      >
        <section
          className="css-w82pwy"
        >
          <h4
            className="css-19ey912"
          >
            More
          </h4>
          <div>
            <amp-accordion
              id="accordion"
            >
              <section
                className="css-w82pwy"
              >
                <h4
                  className="css-19ey912"
                >
                  Apps & Digital Editions
                </h4>
                <div>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    The Economist apps
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    Espresso
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    Global Business Review
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    World in Figures
                  </a>
                </div>
              </section>
            </amp-accordion>
            <amp-accordion
              id="accordion"
            >
              <section
                className="css-w82pwy"
              >
                <h4
                  className="css-19ey912"
                >
                  From The Economist Group
                </h4>
                <div>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    Events
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    Online GMAT prep
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    Online GRE prep
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    Executive Education Navigator
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    Which MBA
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    Jobs Board
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    Learning.ly
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    The Economist Store
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    The Economist Intelligence Unit
                  </a>
                  <a
                    className="css-1xnkqks"
                    href="/"
                  >
                    The Economist Corporate Network
                  </a>
                </div>
              </section>
            </amp-accordion>
            <a
              className="css-1xnkqks"
              href="/"
            >
              About The Economist
            </a>
          </div>
        </section>
      </amp-accordion>
      <a
        className="css-1i6w9mr"
        href="https://econ.st/2bs1gXz"
        role="button"
      >
        Subscribe
      </a>
    </div>
  </section>
</amp-accordion>
`;

exports[`AccordionListItem should match snapshot with normal link 1`] = `
<a
  className="css-1xnkqks"
  href="/"
>
  Link
</a>
`;

exports[`AccordionListItem should match snapshot with subscribeButton 1`] = `
<a
  className="css-1i6w9mr"
  href="https://econ.st/2bs1gXz"
  role="button"
>
  Subscribe
</a>
`;
